cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/deploy/pipeline/DeployDatasetModulesStage.java

```diff
@@ -64,12 +64,12 @@ public void process(ApplicationDeployable input) throws Exception {
         // note: we can deploy module or create module from Dataset class
         // note: it seems dangerous to instantiate dataset module here, but this will be fine when we move deploy into
         //       isolated user's environment (e.g. separate yarn container)
-        Id.DatasetModule moduleId = Id.DatasetModule.from(input.getId().getNamespaceId(), moduleName);
+        Id.DatasetModule moduleId = Id.DatasetModule.from(input.getId().getNamespace(), moduleName);
         if (DatasetModule.class.isAssignableFrom(clazz)) {
           datasetFramework.addModule(moduleId, (DatasetModule) clazz.newInstance());
         } else if (Dataset.class.isAssignableFrom(clazz)) {
           // checking if type is in already
-          if (!datasetFramework.hasType(clazz.getName())) {
+          if (!datasetFramework.hasSystemType(clazz.getName())) {
```

Comment:
don't we still need to check that it doesn't already have that type in its namespace in addition to checking the system?